reference/src/taimpl/src/ta_sa_svp_buffer_copy_blocks.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2020-2022 Comcast Cable Communications Management, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "client_store.h"
+#include "log.h"
+#include "ta_sa.h"
+
+sa_status ta_sa_svp_buffer_copy_blocks(
+        sa_svp_buffer out,
+        sa_svp_buffer in,
+        sa_svp_block* blocks,
+        size_t blocks_length,
+        ta_client client_slot,
+        const sa_uuid* caller_uuid) {
+
+    if (caller_uuid == NULL) {
+        ERROR("NULL caller_uuid");
+        return SA_STATUS_NULL_PARAMETER;
+    }
+
+    if (blocks == NULL) {
+        ERROR("NULL blocks");
+        return SA_STATUS_NULL_PARAMETER;
+    }
+
+    sa_status status;
+    client_store_t* client_store = client_store_global();
+    client_t* client = NULL;
+    svp_store_t* svp_store = NULL;
+    svp_t* out_svp = NULL;
+    svp_t* in_svp = NULL;
+    do {
+        status = client_store_acquire(&client, client_store, client_slot, caller_uuid);
+        if (status != SA_STATUS_OK) {
+            ERROR("client_store_acquire failed");
+            break;
+        }
+
+        svp_store = client_get_svp_store(client);
+        status = svp_store_acquire_exclusive(&out_svp, svp_store, out, caller_uuid);
+        if (status != SA_STATUS_OK) {
+            ERROR("svp_store_acquire_exclusive failed");
+            break;
+        }
+
+        status = svp_store_acquire_exclusive(&in_svp, svp_store, in, caller_uuid);
+        if (status != SA_STATUS_OK) {
+            ERROR("svp_store_acquire_exclusive failed");
+            break;
+        }
+
+        svp_buffer_t* out_svp_buffer = svp_get_buffer(out_svp);
+        svp_buffer_t* in_svp_buffer = svp_get_buffer(in_svp);
+        for (size_t i = 0; i < blocks_length; i++) {
+            if (!svp_copy(out_svp_buffer, blocks[i].out_offset, in_svp_buffer, blocks[i].in_offset, blocks[i].length)) {
+                ERROR("svp_copy failed");
+                status = SA_STATUS_BAD_SVP_BUFFER;
```

Comment:
Recommend avoiding general terms like good or bad in naming and use terms like invalid.

Reply:
Done. This was a really big change by the way.